spec/index.adoc

```diff
@@ -1658,19 +1658,20 @@ It is important to note the following for the usage of the `POST` method:
 
 For security reasons, only HTTPS connections are allowed (URLs must start with `https://`). The IC uses industry-standard root CA lists to validate certificates of remote web servers.
 
-Each request can specify the maximal expected size for the response from the remote HTTP server. The upper limit on the size of a response defaults to `2MiB` if no maximal size value is specified.
-An error will be returned when the response is larger than the maximal size.
-The `2MiB` size limit also applies to the value returned by the `transform` function.
+Each request can specify the maximal expected size for the response body from the remote HTTP server. The upper limit on the size of a response body defaults to `2MiB` if no maximal size value is specified.
+An error will be returned when the response body is larger than the maximal size.
 
 The following parameters should be supplied for the call:
 
-- `url` - the requested URL. The URL may specify a custom port number.
-- `max_response_bytes` - optional, specifies the maximal size of the response in bytes. Any value less than or equal to `2MiB` is accepted. The call will be charged based on this parameter. If not provided, the maximum of `2MiB` will be used.
+- `url` - the requested URL. The URL must be valid according to https://www.ietf.org/rfc/rfc3986.txt[RFC-3986] and its length must not exceed 65534. The URL may specify a custom port number.
+- `max_response_bytes` - optional, specifies the maximal size of the response *body* in bytes. Any value less than or equal to `2MiB` is accepted. The call will be charged based on this parameter. If not provided, the maximum of `2MiB` will be used.
```

Comment:
Is it really just accounting for the body size? @tthebst

Reply:
I've tested that in this [PR](https://github.com/dfinity/ic-hs/pull/100).

Reply:
Currently yes. The limit is only used to limit the response that we receive over the wire. It is used to account for the bandwidth cost. But thinking about it it might also be a good idea to use it to limit the size of the transform.

Reply:
> The limit is only used to limit the response that we receive over the wire.

The headers and status line are also received over the wire, but not accounted for.